profiler/profiler.go

```diff
@@ -342,6 +344,12 @@ func (p *profiler) collect(ticker <-chan time.Time) {
 		for _, prof := range completed {
 			bat.addProfile(prof)
 		}
+		// TODO(fg) this should be done inside of the CPUProfile.Collect() method,
+		// otherwise we'll get the wrong numbers when the CPU duration is different
+		// from the profile duration. Fixing this will require a larger refactor
+		// Meanwhile the incorrect implementation here is okay for customers using
+		// the default profiler settings.
```

Comment:
No refactor necessary :)  In #1485, I arranged for CPU profiling to always finish at the end of the profiling period, even for shorter durations. So you can read the counts here, and you can reset the endpoint counter before CPU profiling is started ([here](https://github.com/DataDog/dd-trace-go/blob/236f16d99b9921aa3dda1a3a1a0cbc2927f17b2a/profiler/profile.go#L92)) rather than in this function.

Are you planning to use these counts to also do something like allocations per request? I'm wondering what you'd do in that case if the CPU duration is less than a minute, but allocations cover the whole minute.

Reply:
Thanks nick. I guess that works, but it feels pretty hacky 🙈. But I guess I'd be okay with it.

Allocations per request will not work for now since we don't have (pprof) endpoint labels for allocations in Go.

That being said, @mar-kolya should confirm which request counts the runtimes should send when e.g. CPU duration and profiling period are different.

Reply:
As discussed offline: it should be fine to send only requests that started (or ended) during the profiling time. This should average out.